src/main/java/com/twilio/http/TwilioRestClient.java

```diff
@@ -59,7 +68,14 @@ protected TwilioRestClient(Builder b) {
      * @return Response object
      */
     public Response request(final Request request) {
-        request.setAuth(username, password);
+        if (username != null && password != null) {
+            request.setAuth(username, password);
```

Comment:
Maybe we can move get auth string to individual auth strategy class and just fetch and set the value in request class

Reply:
In case if we moved username and password to authstrategy class, we can do that.